src/main/java/uk/gov/hmcts/darts/common/repository/ExternalObjectDirectoryRepository.java

```diff
@@ -111,6 +111,20 @@ List<ExternalObjectDirectoryEntity> findExternalObjectDirectoryByLocation(Extern
                                                                               AnnotationDocumentEntity annotation,
                                                                               CaseDocumentEntity caseDocument);
 
+    @Query(
+        """
+            SELECT eod.id FROM ExternalObjectDirectoryEntity eod
+            WHERE eod.status in :failedStatuses
+            AND eod.externalLocationType = :type
+            AND eod.transferAttempts <= :transferAttempts
+            AND eod.osrUuid is null
+            """
+    )
+    List<Integer> findNotFinishedAndNotExceededRetryInStorageLocationIds(List<ObjectRecordStatusEntity> failedStatuses,
```

Comment:
I can still see in few places the old repository method is used e.g. DETStoArm. If we want the query to return only EOD instead of the entire EOD object then we must make changes everywhere other wise we will be left with inconsistent code.

Reply:
Updated